programs/redstone-sol/src/state.rs

```diff
@@ -0,0 +1,37 @@
+use anchor_lang::prelude::*;
+
+pub type SignerAddress = [u8; 20];
+pub type FeedId = [u8; 32];
+
+#[account]
+#[derive(Default)]
+pub struct PriceData {
+    pub feed_id: FeedId,
+    pub value: u128,
+    pub timestamp: u64,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct DataPoint {
+    pub feed_id: FeedId,
+    pub value: u128,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct DataPackage {
+    pub signer_address: SignerAddress,
+    pub timestamp: u64,
+    pub data_points: Vec<DataPoint>,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct Payload {
+    pub data_packages: Vec<DataPackage>,
+}
+
+pub struct Config {
+    pub signer_count_threshold: u8,
+    pub block_timestamp: u64,
+    pub signers: [SignerAddress; 10],
```

Comment:
generally we'd like to pass any vector here.

and, why there's `10` as the signer count ;p?

Reply:
20b34ef99a5a67c5c0e4d7ba67b2069823f98571